### ISSUE TITLE
programs.git: includes should support a list of conditions

### ISSUE THREAD

**Issue by lilyball** [nix-community/home-manager#4310]
### Description

`programs.git.includes` supports conditional includes. It's not unreasonable for the user to want several conditions on a single include (e.g. several locations, several remotes, etc). Right now the whole include has to be duplicated for each condition which is annoying. It would be convenient if `programs.git.includes.*.condition` accepted a list of strings and duplicated it internally instead. This way I could write e.g.

```nix
{
  programs.git.includes = [{
    condition = [
      "hasconfig:remote.*.url:https://ghe.example.com/**"
      "hasconfig:remote.*.url:ssh://git@ghe.example.com/**"
    ];
    contents.user.email = "myworkemail@example.com";
  }];
}
```

**Comment by mattiasflodin** [comment]
I agree with your need (I've run into the same thing), but IMHO it's a bit of a stretch to expect this intelligence from the package when Git does not support multiple conditions natively. As you say, it's possible to get the behavior you want by duplicating the include. If you don't want to duplicate all of the settings, you can specify a path to a configuration file that you reuse multiple times.

That said, thanks for the example! It wasn't obvious to me from the option descriptions how the includes setting would be used in practice.
